Sources/Classes/core/RealMiniApp.swift

```diff
@@ -361,6 +386,35 @@ internal class RealMiniApp {
     func getMiniAppPreviewInfo(using token: String, completionHandler: @escaping (Result<PreviewMiniAppInfo, MASDKError>) -> Void) {
         previewMiniAppInfoFetcher.fetchPreviewMiniAppInfo(apiClient: miniAppClient, using: token, completionHandler: completionHandler)
     }
+
+    func getCachedMiniApp(appId: String,
+                          version: String,
+                          queryParams: String? = nil,
+                          completionHandler: @escaping (Result<MiniAppDisplayDelegate, MASDKError>) -> Void,
+                          messageInterface: MiniAppMessageDelegate? = nil,
+                          adsDisplayer: MiniAppAdDisplayer? = nil) {
+        if appId.isEmpty {
+            return completionHandler(.failure(.invalidAppId))
+        }
+        if version.isEmpty {
+            return completionHandler(.failure(.invalidVersionId))
+        }
```

Comment:
I'm still not sure we should test version number. @corycaywood, if we try to launch a new version that failed to be cached, should we return the latest version in cache?

Reply:
Is this checking the version id which was passed in the `create` interface?
`versionId` is optional, right? So I think it should check the version id ONLY if it is not `nil`.

Otherwise, if version id is `nil`, then it should return whatever version that is cached.

Reply:
@corycaywood @Climbatize . Understood 👍🏻 . Updated-based feedback.